code/datums/weather/weather_types/solar_flare.dm

```diff
@@ -0,0 +1,50 @@
+/datum/weather/solar_flare
+	name = "solar flare"
+	desc = "An intense blast of light and heat from the sun, affecting all space around the station."
+
+	telegraph_duration = 300 // 30 seconds
+	telegraph_message = null // handled via event announcement
+
+	weather_message = "<span class='userdanger'><i>A solar flare has arrived! Find shelter!</i></span>"
+	weather_overlay = "light_ash"
+	weather_duration_lower = 3000 // 5-15 minutes
+	weather_duration_upper = 9000
+	weather_color = "#ffff00" // bright yellow
+	weather_sound = 'sound/misc/bloblarm.ogg' // also used by radiation storm and SM
+
+	end_duration = 10 // wind_down() does not do anything for this event, so we just trigger end() semi-immediately
+	end_message = null
+	area_type = /area/space // read generate_area_list() as well below
+	protected_areas = list()
+	target_trait = STATION_LEVEL
+	immunity_type = "burn"
+
+/datum/weather/solar_flare/generate_area_list()
+	..()
+	var/list/bonus_areas = list()
+	for(var/V in get_areas(/area/solar))
+		// no, solars in space are NOT a subtype of /area/space.
+		// no, we don't want to re-path every reference to all the subtypes of /area/solar across every map file.
+		// no, we don't want to change /datum/weather/var/area_type into a list as that requires changing every item that touches weather
+		bonus_areas += V
+	for(var/V in bonus_areas)
+		var/area/A = V
+		if(A.z in impacted_z_levels)
+			impacted_areas |= A
+
+/datum/weather/solar_flare/start()
+	..()
+	// Solars produce 40x as much power. 240KW becomes 9.6MW. Enough to cause APCs to arc all over the station if >=2 solars are hotwired.
+	SSsun.solar_gen_rate = 1500 * 40
+
+/datum/weather/solar_flare/weather_act(mob/living/L)
+	L.adjustFireLoss(1)
+	if(prob(10))
+		to_chat(L, "<span class='warning'>The solar flare burns you! Seek shelter!")
+
+/datum/weather/solar_flare/end()
+	if(..())
+		return
+	GLOB.event_announcement.Announce("The solar flare has passed.", "Solar Flare Advisory")
+	// Solars produce 40x as much power. 240KW becomes 9.6MW. Enough to cause APCs to arc all over the station if >=2 solars are hotwired.
```

Comment:
Change this to "Lowers the rate from the 40x increase" or something similar

Reply:
Changed